### PR TITLE
ci(i): Add workflow to ensure cli docs are in sync

### LINKED ISSUE #2165: Assert output of `make docs` is up to date in CI

### PR DESCRIPTION
**Pull request by shahzadlone**
## Relevant issue(s)
Resolves #2165

## Description
- Detects if cli documents need updating.
- If Documentation is up to date the action passes, else fails.
- Run `make docs` or `make docs:cli` to generate the documentation.
- Introducing as a non-required check, but would argue it should be required.

## Tasks
- [x] I made sure the code is well commented, particularly hard-to-understand areas.
- [x] I made sure the repository-held documentation is changed accordingly.
- [x] I made sure the pull request title adheres to the conventional commit style (the subset used in the project can be found in [tools/configs/chglog/config.yml](tools/configs/chglog/config.yml)).
- [x] I made sure to discuss its limitations such as threats to validity, vulnerability to mistake and misuse, robustness to invalidation of assumptions, resource requirements, ...

## How has this been tested?
- Manually testing through action run and by using `act` utility.
- Failed action: https://github.com/sourcenetwork/defradb/actions/runs/9214068142/job/25349425467?pr=2647

Specify the platform(s) on which this was tested:
- WSL2

### CODE REVIEW

File: .github/workflows/check-cli-documentation.yml
Comment (AndrewSisley) [inline .github/workflows/check-cli-documentation.yml:38]: question: Why is this step needed?

todo: If keeping, please document why we want this :)

**Review by fredcarle — APPROVED** [review]
Approving assuming Andy's comment gets resolved.

File: .github/workflows/check-cli-documentation.yml
Comment (shahzadlone) [reply .github/workflows/check-cli-documentation.yml:38]: Incase upon checkout some dirty state exists (i don't think it's going to happen, but just for extra caution). It's be nice to catch that inconsistency.

Happy to remove if there is preference/consensus.

Comment (AndrewSisley) [reply .github/workflows/check-cli-documentation.yml:38]: Okay, I'm somewhat sceptical if a check here is the right place for that, but it is a cheap action and not worth spending time on :)

Please do document this though before merge.

**Review by AndrewSisley — APPROVED** [review]
Approving assuming that the 2nd step is documented before merge - thanks Shahzad :)

**Pull request merged**
